src/arg_max.py

```diff
@@ -0,0 +1,42 @@
+from nada_dsl import *
+import nada_numpy as na
+
+DIM = 10
+
+def argmax(array: na.NadaArray, index_party: Party):
+    # The result (the index of the argmax)
+    result = SecretInteger(Input(name="index", party=index_party))
```

Comment:
This doesn't need to be an input. In line 23 where you do 
```
        result = cond.if_else(current_index, result)
```
since the `cond` and `current_index` are Secret, the result will also be secret.

So what you could do here is just:
```
result = Integer(-1)
```
to initialize it and in line 23 it'll get overwritten (whichever branch is obliviously taken).

Reply:
It will fail if the max value is at index 0, as in `arg_max_test_4`. This will leave `result` not changed. But, there is a workaround, though; we can change the condition at line 19 to be `cond = v >= max_val` instead of `cond = v > max_val`. In this way, we will ensure that `result` will updated accordingly.  

Also, if there are more than one maximum value (i.e., two equal values which are both the maximum among the list), `result` will be set to the index of the last maximum value.